### PR TITLE
feat: add c bindings for variable length multi-exponentiations (PROOF-896)

### PR DESCRIPTION
**Pull request by rnburn**
# Rationale for this change

Add c bindings for variable length multi-exponentiations (follow up to #166 )

# What changes are included in this PR?

* Adds sxt_fixed_vlen_multiexponentiation to the c api for variable length multiexponentiations
* adds code to handle an edge case of multi-exponentiations with no outputs

# Are these changes tested?

Yes

### CODE REVIEW

File: cbindings/fixed_pedersen.t.cc [hunk 1]
```diff
@@ -91,6 +91,40 @@ TEST_CASE("we can compute multi-exponentiations with a fixed set of generators")
...
+    cbn::reset_backend_for_testing();
+    const sxt_config config = {SXT_GPU_BACKEND, 0};
+    REQUIRE(sxt_init(&config) == 0);
+
+    wrapped_handle h{generators.data(), 2};
+    REQUIRE(h.h != nullptr);
+
+    uint8_t scalars[] = {0b1010, 0b0101};
+    unsigned bit_table[] = {3, 1};
+    unsigned lengths[] = {2, 2};
```
Comment (jacobtrombetta) [inline cbindings/fixed_pedersen.t.cc:104]: Should there be a test for variable `lengths`, like `unsigned lengths[] = {1,2}`?

File: cbindings/blitzar_api.h [hunk 2]
```diff
@@ -642,6 +642,33 @@ void sxt_fixed_packed_multiexponentiation(void* res, const struct sxt_multiexp_h
...
 
+/**
+ * Compute a varying lengthing multiexponentiation of scalars in packed format using a handle to
+ * pre-specified generators.
+ *
+ * On completion `res` contains an array of size `num_outputs` for the multiexponentiation
+ * of the given `scalars` array.
+ *
+ * An entry output_bit_table[output_index] specifies the number of scalar bits used for
+ * output_index and output_lengths[output_index] specifies the length used for output_index.
```
Comment (jacobtrombetta) [inline cbindings/blitzar_api.h:653]: Does `length` refer to the number of pre-specified generators are used during multiexponentiation for the specific `output_index`?

File: sxt/cbindings/backend/computational_backend_utility.cc [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  SXT_DEBUG_ASSERT(output_lengths.size() == num_outputs);
+
+  unsigned output_bit_sum = 0;
+  unsigned n = 0;
+  unsigned prev_len = 0;
+  for (unsigned output_index = 0; output_index < num_outputs; ++output_index) {
+    auto width = output_bit_table[output_index];
+    SXT_RELEASE_ASSERT(width > 0, "output bit width must be positive");
+    auto len = output_lengths[output_index];
+    SXT_RELEASE_ASSERT(len >= prev_len, "output lengths must be sorted in ascending order");
```
[the remark this answers is not in the document]
Comment (rnburn) [reply sxt/cbindings/backend/computational_backend_utility.cc:41]: Added a note. Thank you.

File: cbindings/blitzar_api.h (hunk 2 above)
Comment (rnburn) [reply cbindings/blitzar_api.h:653]: Yes.

File: cbindings/fixed_pedersen.t.cc (hunk 1 above)
Comment (rnburn) [reply cbindings/fixed_pedersen.t.cc]: I modified the tests to be of varying length.

**Review by jacobtrombetta — APPROVED** [review]

**Pull request merged**